app/commands/v2/put_content.rb

```diff
@@ -42,11 +44,19 @@ def link_diff_between(old_links, new_links)
         old_links - new_links
       end
 
-      def prepare_content_with_base_path
+      def remove_previous_path_reservations
+        to_discard = previous_drafted_edition&.base_path
+        return if to_discard.blank? || to_discard == payload[:base_path]
+        return if to_discard == previously_published_edition.previous_base_path
```

Comment:
I found this a bit hard to understand, since there may be other historical paths we care about. Are we really asking `will_redirect?`

Reply:
The question isn't whether it will redirect it's asking is the path already in use on published content. Your question has prompted me to wonder whether I've done this right though because there are scenarios where this could discard a path reservation still used by an app (for example you create a new document to replace an unpublished one) so it might be better to instead query where there is a live content item using that base_path.

Reply:
I'm still not sure I understand. While the new code is self-explanatory (thanks!), I'm still wondering about the behaviour of this with respect to redirects on previously live (superseded) editions: this looks like [it will discard them](https://github.com/alphagov/publishing-api/blob/master/app/models/edition.rb#L165) (and therefore allow them to be overridden)? Perhaps we should increase the scope to be "NOT content_store = 'draft'".

Reply:
Gotcha, I don't think it will prove problematic - I'll try explain my reasoning. Once any redirects related to a superseding are live, they will have the base_path and same publishing app of the content that was superseded, thus still meet the criteria in the query. In theory, and assuming everything now and previously works properly, every base_path ever publicly used on GOV.UK should have an Edition with a content_store of live in some state.

I don't think it should go to the degree of asking NOT content_store draft, as that means the past can interfere with the process in subtle, surprising ways (for example you happen to put content an old path that was once used in 2016 and has nothing else and this process fails). In general, I also think if we end up with a base_path not marked as being on live content store then it's effectively historical and fair game to be taken.

Does that help at all?

Reply:
> they will have the base_path and same publishing app of the content that was superseded

Ah, that's good to know. Perhaps there's still a loophole here for a chain of redirects, but that also seems (a) unlikely and (b) low impact; at worst, we might lead people to mildly irrelevant content.